src/main/scala/org/broadinstitute/dsde/workbench/sam/util/LdapSupport.scala

```diff
@@ -59,17 +59,24 @@ trait LdapSupport extends DirectorySubjectNameSupport {
       case Failure(regrets) => throw regrets
     }
 
-  protected def getAttribute(results: Entry, key: String): Option[String] =
-    Option(results)
-      .flatMap { searchResultEntries => Option(searchResultEntries.getAttribute(key)) }
-      .map(_.getValue)
+  protected def getAttribute(results: Entry, key: String): Option[String] = {
+    for {
+      searchResultEntries <- Option(results)
+      attributes <- Option(searchResultEntries.getAttribute(key))
+    } yield {
+     attributes.getValue
```

Comment:
Perhaps a nit, but the naming choice to use plurals for `searchResultEntries` and `attributes` made me need to think twice and then confused me with the method below for a second.  I would prefer using variable names that are singular in this method.